docs/mutations.md

```diff
@@ -152,136 +152,4 @@ If the mutation failed, changes performed by an optimistic update are overwritte
-
-```python
-def resolve_create_discussion(_, info, input):
-    clean_input = {
-        "category": input["category"],
-        "title": input["title"],
-        "is_announcement": input.get("isAnnouncement"),
-        "is_closed": input.get("isClosed"),
-    }
-
-    try:
-        return {
-            "status": True,
-            "discussion": create_new_discussion(info.context, clean_input),
-        }
-    except ValidationError as err:
-        return {
-            "status": False,
-            "error": err,
-        }
-```
-
-Another advantage of `input` types is that they are reusable. If we later decide to implement another mutation for updating the `Discussion`, we can do it like this:
-
-```graphql
-type Mutation {
-    createDiscussion(input: DiscussionInput!): DiscussionPayload
-    updateDiscussion(discussion: ID!, input: DiscussionInput!): DiscussionPayload
-}
-
-input DiscussionInput {
-    category: ID!
-    title: String!
-    isAnnouncement: Boolean
-    isClosed: Boolean
-}
-```
-
-Our `updateDiscussion` mutation will now accept two arguments: `discussion` and `input`:
-
-```python
-def resolve_update_discussion(_, info, discussion, input):
-    clean_input = {
-        "category": input["category"],
-        "title": input["title"],
-        "is_announcement": input.get("isAnnouncement"),
-        "is_closed": input.get("isClosed"),
-    }
-
-    try:
-        return {
-            "status": True,
-            "discussion": update_discussion(info.context, discussion, clean_input),
-        }
-    except ValidationError as err:
-        return {
-            "status": False,
-            "error": err,
-        }
-```
-
-You may wonder why you would want to use `input` instead of reusing an already-defined type. This is because input types provide some guarantees that regular objects don't: they are serializable, and they don't implement interfaces or unions. However, input fields are not limited to scalars. You can create fields that are lists, or even reference other inputs:
-
-```graphql
-input PollInput {
-    question: String!,
-    options: [PollOptionInput!]!
-}
-
-input PollOptionInput {
-    label: String!
-    color: String!
-}
-```
-
-Lastly, take note that inputs are not specific to mutations. You can create inputs to implement complex filtering in your `Query` fields.
-
-> **Note:** you can decorate your resolvers with [`convert_kwargs_to_snake_case`](api-reference.md#convert_kwargs_to_snake_case) to convert arguments and inputs names from `camelCase` to `snake_case`.
+> There is no requirement for every mutation to have its own `Payload` type. `login` and `logout` mutations can both define `LoginResult` as their return type. It is up to the developer to decide how generic or specific mutation results should be.
```

Comment:
```suggestion
> There is no requirement for every mutation to have its own `Result` type. `login` and `logout` mutations can both define `LoginResult` as their return type. It is up to the developer to decide how generic or specific mutation results should be.
```

Also in the example above (I can't comment on collapsed lines 🤔) `userMutationPayload` -> `userMutationResult`